src/mame/sun/sun4.cpp

```diff
@@ -609,29 +642,36 @@ class sun4_base_state : public driver_device
 	required_device<ram_device> m_ram;
 	required_memory_region m_rom;
 
-	uint8_t m_auxio;
-	uint32_t m_counter[4];
-	uint32_t m_dma[4];
+	u8 m_auxio;
+	u32 m_timer_count[2];
+	u32 m_timer_limit[2];
+	u32 m_dma_ctrl;
+	u32 m_dma_addr;
+	u32 m_dma_count;
 	bool m_dma_irq;
 	bool m_dma_tc_read;
-	uint32_t m_dma_pack_register;
-	int m_scsi_irq;
-	int m_fdc_irq;
+	u32 m_dma_pack_register;
+	bool m_scsi_irq;
+	bool m_fdc_irq;
 
-	uint8_t m_irq_reg;    // IRQ control
-	uint8_t m_scc1_int, m_scc2_int;
+	u8 m_irq_reg;    // IRQ control
+	u8 m_scc_int[2];
 
-	attotime m_c0_last_read_time;
-	attotime m_c1_last_read_time;
-	emu_timer *m_c0_timer;
-	emu_timer *m_c1_timer;
+	emu_timer *m_timer[2];
 
 	void dma_check_interrupts();
 	void dma_transfer();
 	void dma_transfer_write();
 	void dma_transfer_read();
 
 	void fcodes_command(int ref, const std::vector<std::string_view> &params);
+
+	static constexpr u32 LOG_TIMER_COUNT_WRITES[2] = { LOG_TMR0_COUNT_WRITES, LOG_TMR1_COUNT_WRITES };
+	static constexpr u32 LOG_TIMER_LIMIT_WRITES[2] = { LOG_TMR0_LIMIT_WRITES, LOG_TMR1_LIMIT_WRITES };
+	static constexpr u32 LOG_TIMER_COUNT_READS[2] = { LOG_TMR0_COUNT_READS, LOG_TMR1_COUNT_READS };
+	static constexpr u32 LOG_TIMER_LIMIT_READS[2] = { LOG_TMR0_LIMIT_READS, LOG_TMR1_LIMIT_READS };
+	static constexpr int TIMER_IRQ_LINES[2] = { SPARC_IRQ10, SPARC_IRQ14 };
+	static constexpr u8 TIMER_IRQ_MASKS[2] = { 0x21, 0x81 };
```

Comment:
Can you please make these `static constexpr inline`?  It will avoid the possibility of linker errors if the compiler decides to generate references to them.